### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by olawale-o**
- Create class Person with the following:
  - Instance vars: @id, @name, and @age.
  - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have        default values of "Unknown" and true.
  - Getters for @id, @name, and @age.
  - Setters for @name and @age.
  - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
  - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Create class Student with the following:
  - Inherits from Person.
  - Constructor extends parent's constructar by adding @classroom and a parameter for it.
  - Method play_hooky that returns "¯\(ツ)/¯".
- Create class Teacher with the following:
  - Inherits from Person.
  - Constructor extends parent's constructar by adding @specialization and a parameter for it.
  - Override can_use_services? so it always returns true.
- Each class was saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Person
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @id = 1
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (Emmyn5600) [inline person.rb:10]: - [ ] By using this way will probably throw an error saying: NoMethodError: undefined method `id, name, age and parent_permission`.

`NoMethodError` is the error you get when you ask for the value of `age, name and parent_permission` without the proper setup.

The solution, you can define your own method like this:
class Person 
  def name
    @name
  end
end

However, this should be hard/complicated while we need to implement more methods.

So, there is a better way to define this kind of method? like `attr_accessor`, `attr_reader` and `attr_writer`.

class Person
 attr_accessor :name
 def initialize(name)
   @name = name
 end
end

For more details or explanation, refer to this [link](https://mixandgo.com/learn/ruby_attr_accessor_attr_reader_attr_writer) and this [one](https://stackoverflow.com/questions/5046831/why-use-rubys-attr-accessor-attr-reader-and-attr-writer)

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  def can_use_services?
+    return true if is_of_age? || @parent_permission
+
+    false
+  end
+
+  private :is_of_age?
+end
```
Comment (Emmyn5600) [inline person.rb]: - [ ] Kindly,  Try to fix rubocop errors without disabling it, this makes your code looks unprofessional and unclear, you can run `rubocop` to check for errors and `rubocop --fix` to fix them. check the screenshot below:

--------rubocop------------

ruby
--------rubocop --fix ------------

ruby-1

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

### Hi, @olawale-o 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

   ###  Changes required :recycle: 
   
    -  Add a descriptive Readme file
    -  Please check the comment under the review too

    Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 2 above)
Comment (olawale-o) [reply person.rb]: Done

File: person.rb (hunk 1 above)
Comment (olawale-o) [reply person.rb:10]: Done

**Comment by Davidosky007** [comment]
## Hi @olawale-o 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by olawale-o** [comment]
Hi @Emmyn5600 
Thank you for your time and effort in pointing out these corrections

**Comment by olawale-o** [comment]
Hi @Davidosky007 
Thank you for your time and effort on the review

**Pull request merged**